waspc/data/Generator/templates/react-app/src/api.ts

```diff
@@ -40,6 +51,20 @@ api.interceptors.response.use(undefined, (error) => {
   return Promise.reject(error)
 })
 
+// This handler will run on other tabs (not the active one), and will
```

Comment:
Aha, so this basically listens for changes to storage coming from somewhere else and then updates the state based on that. And main purpose of this is to detected changes in other tabs.

So what happens when somebody calls setAuthToken -> won't that trigger this listener also, and then it will get triggered redundantly? So we will already set `authToken` variable in this module, and we will already emit the event 'authToken.set', but then this will trigger again and those things will happen again? I guess it doesn't hurt, but it is duplicate work + our event is being sent out twice which is a bit weird.

Reply:
No, the active tab will call the API functions, but won't get this handler. This only runs in the non-active windows (aka the one _not_ calling API functions).

Reply:
Oh interesting! What magic is that? It doesn't emit that event to itself?

Reply:
By design it seems https://developer.mozilla.org/en-US/docs/Web/API/Window/storage_event

> Note: This won't work on the same page that is making the changes — it is really a way for other pages on the domain using the storage to sync any changes that are made. Pages on other domains can't access the same storage objects.